### PR TITLE
Remove cuda compat

### PR DESCRIPTION
**Pull request by jahaniam**
*Issue #475 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: micro-benchmarks/nccl-tests/nccl-tests.Dockerfile [hunk 1]
```diff
@@ -45,7 +45,7 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
     pkg-config \
     python3-distutils \
     vim \
-    && apt-get install -y --upgrade ${NV_CUDA_COMPAT_PACKAGE}
+    && apt-get install -y --upgrade
```
Comment (mhuguesaws) [inline micro-benchmarks/nccl-tests/nccl-tests.Dockerfile:48]:
```suggestion
RUN apt-get purge -y cuda-compat-*
```

File: micro-benchmarks/nccl-tests/nccl-tests.Dockerfile [hunk 2]
```diff
@@ -45,7 +45,7 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
     pkg-config \
     python3-distutils \
     vim \
```
Comment (mhuguesaws) [inline micro-benchmarks/nccl-tests/nccl-tests.Dockerfile:47]:
```suggestion
    vim
```

**Review by mhuguesaws — COMMENTED** [review]
Left suggestions.

File: micro-benchmarks/nccl-tests/nccl-tests.Dockerfile (hunk 1 above)
Comment (mhuguesaws) [reply micro-benchmarks/nccl-tests/nccl-tests.Dockerfile:48]: `cuda-compat` comes from the base container, it needs to be explicitly removed.

Comment (jahaniam) [reply micro-benchmarks/nccl-tests/nccl-tests.Dockerfile:48]: Sure, but we don't need to remove it too if we dont add it to LIBRARY_PATH.

**Comment by jahaniam** [comment]
> Left suggestions.

@mhuguesaws sounds good. Your suggestions looked good and applied them. Please review.

**Review by mhuguesaws — APPROVED** [review]
LGTM

**Pull request merged**